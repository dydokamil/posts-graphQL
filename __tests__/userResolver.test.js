// jest.mock('axios')

const mongoose = require('mongoose')

const { MONGO_URL_DEV } = require('../consts')
const User = require('../models/user')
const Post = require('../models/post')
const { Query, Mutation } = require('../resolvers')

describe('user resolver', () => {
  const userData = {
    _id: '5abba8e47af4d91c259e12ef',
    password: 'test',
    username: 'User1',
    email: 'User1@gql.com',
    createdAt: '2018-12-12T13:00:00',
    lastLogin: '2018-12-12T03:14:07',
    posts: ['5abba8e47af4d91c259e12ee']
  }

  const postData = {
    _id: '5abba8e47af4d91c259e12ee',
    author: userData._id,
    createdAt: '2018-10-10T13:00:00',
    editedAt: '2018-10-10T13:00:00',
    message: 'Some message'
  }

  beforeAll(async () => {
    await mongoose.connect(MONGO_URL_DEV)
    // clear the database
    await User.removeUsers()
    await Post.removePosts()
  })

  beforeEach(async () => {
    // const userInstance = new User(userData)
    // await userInstance.save()
    // const postInstance = new Post(postData)
    // await postInstance.save()
  })

  afterEach(async () => {
    await User.removeUsers()
    await Post.removePosts()
  })

  afterAll(async done => {
    await mongoose.disconnect(done)
  })

  it('should respond with users', () => {
    const userInstance = new User(userData)
    return userInstance.save().then(() => {
      return Query.users().then(users => {
        expect(users.length).toBe(1)
        expect(users).toMatchSnapshot()
      })
    })
  })

  it('should respond with a user', () => {
    const { _id } = userData

    const userInstance = new User(userData)
    const postInstance = new Post(postData)

    return userInstance.save().then(() => {
      return postInstance.save().then(() => {
        return Query.user({}, { _id }).then(user => {
          expect(user._id.equals(_id)).toBeTruthy()
          expect(user).toMatchSnapshot()
        })
      })
    })
  })

  it('should create a user', () => {
    const username = 'User2'
    const email = 'User2@user.com'
    const password = 'test'

    return Mutation.createUser({}, { username, email, password }).then(user => {
      expect(user.username).toEqual(username)
      expect(user.email).toEqual(email)
      expect(user.password).not.toBe(password)
    })
  })
})
