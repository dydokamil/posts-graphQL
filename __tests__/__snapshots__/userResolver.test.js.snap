// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user resolver should change user password 1`] = `[Error: Password incorrect.]`;

exports[`user resolver should respond with a user 1`] = `
Object {
  "__v": 0,
  "_id": "5abba8e47af4d91c259e12ef",
  "createdAt": 1523390727,
  "email": "User1@gql.com",
  "lastLogin": 1523390727,
  "password": "test",
  "posts": Array [
    Object {
      "__v": 0,
      "_id": "5abba8e47af4d91c259e12ee",
      "author": "5abba8e47af4d91c259e12ef",
      "createdAt": 1523390727,
      "editedAt": 1523390727,
      "message": "Some message",
    },
  ],
  "subjects": Array [],
  "username": "User1",
}
`;

exports[`user resolver should respond with users 1`] = `
Array [
  Object {
    "__v": 0,
    "_id": "5abba8e47af4d91c259e12ef",
    "createdAt": 1523390727,
    "email": "User1@gql.com",
    "lastLogin": 1523390727,
    "password": "test",
    "posts": Array [],
    "subjects": Array [],
    "username": "User1",
  },
]
`;
